### ISSUE TITLE
Fix Request: false positive and missed case in UnresolvedType

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#581]
PMD typeIs seems not able to resolve in the following case:
```
import lombok.Builder;
import com.fasterxml.jackson.annotation.JsonFormat;
import java.time.LocalDateTime;

@Builder
public record UnresolvedTypeTryout (
            Long documentId,
            @JsonFormat(pattern = UnresolvedTypeTryout.DATE_TIME_PATTERN) // works fine, can resolve
            LocalDateTime requestedOn,
            @JsonFormat(pattern = DATE_TIME_PATTERN) // <-- false positive, cannot resolve
            LocalDateTime refusedOn
) {
        private static final String DATE_TIME_PATTERN = "yyyy-MM-dd'T'HH:mm:ss";
}

```

**Comment by jborgers** [comment]
Spotted a missed case, if slf4j annotation is used, any AmbiguousName is ignored. 
```
import lombok.extern.slf4j.Slf4j;
@Slf4j
record IssueRecord(String name) {
    public void foo(){
        unknownType.doIt(); // bad, missed case
        log.debug("something");
    }
}
```
